tests/basis_test.py

```diff
@@ -1,330 +1,1017 @@
+    assert numpy.shape(f.derivative(xs7)) == numpy.shape(xs7)
+    assert numpy.allclose(f.derivative(xs7), get_answer(xs7))
 
 
 def test_cheb_derivative_invalid_input():
-    """Test call raises error if input is outside domain [-1,1]"""
-    f = ChebyshevFirstKind(4)
+    """Test call raises error if input is outside domain [-1, 1]"""
+    f = smolyay.basis.ChebyshevFirstKind(4)
     with pytest.raises(ValueError):
         f.derivative(2)
     with pytest.raises(ValueError):
         f.derivative(-2)
-
-
-def test_cheb_2nd_initial_zero():
-    """test degree of zero"""
-    f = ChebyshevSecondKind(0)
-    assert f.n == 0
-    assert f.points == [0]
-
-
-def test_cheb_2nd_initial_1():
-    """test initial when degree is 1"""
-    f = ChebyshevSecondKind(1)
-    assert f.n == 1
-    assert f.points == [0]
-
-def test_cheb_2nd_initial_2():
-    """test initial when degree is 2"""
-    f = ChebyshevSecondKind(2)
-    assert f.n == 2
-    assert numpy.allclose(f.points,[-0.5, 0.5],atol=1e-10)
-
-def test_cheb_2nd_initial_3():
-    """test initial when degree is 3"""
-    expected_points = [-1/numpy.sqrt(2), 0, 1/numpy.sqrt(2)]
-    f = ChebyshevSecondKind(3)
-    assert f.n == 3
-    assert numpy.allclose(f.points,expected_points,atol=1e-10)
-
-def test_cheb_2nd_initial_7():
-    """test initial when degree is 7"""
-    expected_points = [-numpy.sqrt(numpy.sqrt(2)+1)/(2**0.75),
-            -1/numpy.sqrt(2), -numpy.sqrt(numpy.sqrt(2)-1)/(2**0.75), 0,
-            numpy.sqrt(numpy.sqrt(2)-1)/(2**0.75), 1/numpy.sqrt(2),
-            numpy.sqrt(numpy.sqrt(2)+1)/(2**0.75)]
-    f = ChebyshevSecondKind(7)
-    assert f.n == 7
-    assert numpy.allclose(f.points,expected_points,atol=1e-10)
-
-def test_cheb_2nd_call_degree_0_1():
-    """Chebyshev polynomial degree 0 is always 1 and degree 1 returns 2*input"""
-    f0 = ChebyshevSecondKind(0)
-    f1 = ChebyshevSecondKind(1)
-    for i in [-1, -0.5, 0, 0.5, 1]:
-        assert f0(i) == 1
-        assert f1(i) == i*2
-
-def test_cheb_2nd_call_random_points():
-    """Test chebyshev polynomial at some degree at some input"""
-    numpy.random.seed(567)
-    ns = numpy.random.randint(20,size = 20)
-    xs = numpy.random.rand(20) * 2 - 1
-    for n,x in zip(ns,xs):
-        f = ChebyshevSecondKind(n)
-        assert numpy.isclose(f(x),special.eval_chebyu(n,x))
-
-def test_cheb_2nd_call_random_points_multi_input():
-    """Test that chebyshev polynomial call handles multiple x inputs"""
-    numpy.random.seed(567)
-    xs = numpy.random.rand(20) * 2 - 1
-    f0 = ChebyshevSecondKind(0)
-    f1 = ChebyshevSecondKind(1)
-    fn = ChebyshevSecondKind(5)
-    assert numpy.allclose(f0(xs),special.eval_chebyu(0,xs))
-    assert numpy.allclose(f1(xs),special.eval_chebyu(1,xs))
-    assert numpy.allclose(fn(xs),special.eval_chebyu(5,xs))
+    with pytest.raises(ValueError):
+        f.derivative([[0.5, 0.7, 3, 0.8], [0.5, 0.7, 0.8, -0.5]])
+
+
+def test_cheb_2nd_initial():
```

Comment:
Are the ChebyshevSecondKind tests all basically the same as first kind? If yes, then you could make the first parameter of the tests be a basic function rather than a degree to reduce duplication!